src/ResourceManagement/BatchAI/BatchAIClustersImpl.cs

```diff
@@ -44,37 +59,58 @@ protected override IBatchAICluster WrapModel(ClusterInner inner)
             {
                 return null;
             }
-            return new BatchAIClusterImpl(inner.Name, inner, Manager);
+            return new BatchAIClusterImpl(inner.Name, workspace, inner);
         }
+        
+        public IClustersOperations Inner => workspace.Manager.Inner.Clusters;
 
-        protected override async Task<ClusterInner> GetInnerByGroupAsync(string groupName, string name, CancellationToken cancellationToken)
+        public IEnumerable<IBatchAICluster> List()
         {
-            return await Inner.GetAsync(groupName, name, cancellationToken);
+            return WrapList(Extensions.Synchronize(() => Inner.ListByWorkspaceAsync(workspace.ResourceGroupName, workspace.Name))
+                .AsContinuousCollection(link => Extensions.Synchronize(() => Inner.ListByWorkspaceNextAsync(link))));
         }
 
-        protected override async Task DeleteInnerByGroupAsync(string groupName, string name, CancellationToken cancellationToken)
+        public async Task<IPagedCollection<IBatchAICluster>> ListAsync(bool loadAllPages = true, CancellationToken cancellationToken = new CancellationToken())
         {
-            await Inner.DeleteAsync(groupName, name, cancellationToken);
+            return await PagedCollection<IBatchAICluster, ClusterInner>.LoadPage(
+                async (cancellation) => await Inner.ListByWorkspaceAsync(workspace.ResourceGroupName, workspace.Name, cancellationToken: cancellation),
+                Inner.ListByWorkspaceNextAsync,
+                WrapModel, loadAllPages, cancellationToken);
         }
 
-        protected override async Task<IPage<ClusterInner>> ListInnerAsync(CancellationToken cancellationToken)
+        public IBatchAICluster GetById(string id)
         {
-            return await Inner.ListAsync(cancellationToken: cancellationToken);
+            ResourceId resourceId = ResourceId.FromString(id);
+            return WrapModel(Extensions.Synchronize(() => workspace.Manager.Inner.Clusters.GetAsync(resourceId.ResourceGroupName, workspace.Name, resourceId.Name)));
         }
```

Comment:
instead of reimplementing you can just use `return Extensions.Synchronize(() =>GetByIdAsync(id))`